### DISCUSSION TITLE
turn off user presence

### DISCUSSION THREAD

**Discussion by shimritd** [Yubico/libfido2#656] [Q&A] [answered]
Hi,

When i turn off user presence and use the hmac-secret extension, i get an error the UP is required.
this behavior happens only with Yubico's keys. 
Is there a way to solve this?
thanks!

**Comment by LDVG** [comment, marked as answer]
Hi,

Per the CTAP 2.1 specification,[ § 12.5 HMAC Secret Extension, authenticatorGetAssertion additional behaviors](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#sctn-hmac-secret-extension), the authenticator should return an error if UP is set to false. This is enforced on the authenticator level.
